src/rollup/index.ts

```diff
@@ -488,36 +467,78 @@ export default function rollup(
 
 					timeStart('GENERATE', 1);
 
-					const generated: { [chunkName: string]: SourceDescription } = {};
-
-					const promise = Promise.all(
-						Object.keys(bundle).map(chunkName => {
-							const chunk = bundle[chunkName];
-							return chunk.render(outputOptions).then(rendered => {
-								timeEnd('GENERATE', 1);
+					const generated: { [chunkName: string]: OutputChunk } = {};
 
-								graph.plugins.forEach(plugin => {
-									if (plugin.ongenerate) {
-										const bundle = chunks[chunkName];
-										plugin.ongenerate(assign({ bundle }, outputOptions), rendered);
-									}
-								});
+					let preserveModulesBase: string;
+					if (inputOptions.experimentalPreserveModules)
+						preserveModulesBase = commondir(chunks.map(chunk => chunk.entryModule.id));
+					let existingNames = Object.create(null);
 
-								generated[chunkName] = rendered;
-							});
+					const promise = createAddons(graph, outputOptions)
+						.then(addons => {
+							// first pre-render all chunks
+							// then name all chunks
+							return (
+								Promise.resolve()
+									.then(() => {
+										chunks.forEach(chunk => chunk.preRender(outputOptions));
+
+										chunks.forEach(chunk => {
+											if (inputOptions.experimentalPreserveModules) {
+												chunk.generateNamePreserveModules(preserveModulesBase);
+											} else {
+												let pattern;
+												if (chunk.isEntryModuleFacade) {
+													pattern = outputOptions.entryNames || '[alias].js';
+												} else {
+													pattern = outputOptions.chunkNames || '[alias]-[hash].js';
```

Comment:
As discussed, maybe we should remove the hash from the default here.

Reply:
This hash case is for chunks that won't be loaded by users only - `chunk-sd8fs9f.js`. Entry points are still `entry.js`, effectively only using the hashes on the "internal chunk names".

Perhaps in the manual chunks PR we could discuss an option here to remove the hashes by default - either by separating `manualChunkNames` or by making the alias itself include the hash, although that kind of defeats the nice mental model :P

I thought we could keep it for these cases over using the numbering (`chunk.js`, `chunk2.js`, ...), as if anything hashes probably make it easier to keep track of chunk and chunk changes for users anyway.